keps/sig-api-machinery/20190724-http-status-check.md

```diff
@@ -0,0 +1,204 @@
+
+**ACTION REQUIRED:** In order to merge code into a release, there must be an issue in [kubernetes/enhancements] referencing this KEP and targeting a release milestone **before [Enhancement Freeze](https://github.com/kubernetes/sig-release/tree/master/releases)
+of the targeted release**.
+
+For enhancements that make changes to code or processes/procedures in core Kubernetes i.e., [kubernetes/kubernetes], we require the following Release Signoff checklist to be completed.
+
+Check these off as they are completed for the Release Team to track. These checklist items _must_ be updated for the enhancement to be released.
+
+- [ ] kubernetes/enhancements issue in release milestone, which links to KEP (this should be a link to the KEP location in kubernetes/enhancements, not the initial KEP PR)
+- [ ] KEP approvers have set the KEP status to `implementable`
+- [ ] Design details are appropriately documented
+- [ ] Test plan is in place, giving consideration to SIG Architecture and SIG Testing input
+- [ ] Graduation criteria is in place
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+**Note:** Any PRs to move a KEP to `implementable` or significant changes once it is marked `implementable` should be approved by each of the KEP approvers. If any of those approvers is no longer appropriate than changes to that list should be approved by the remaining approvers and/or the owning SIG (or SIG-arch for cross cutting KEPs).
+
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://github.com/kubernetes/enhancements/issues
+[kubernetes/kubernetes]: https://github.com/kubernetes/kubernetes
+[kubernetes/website]: https://github.com/kubernetes/website
+
+## Summary
+
+Current HTTP probe is not sufficient for check HTTP service running correct or not. In some
+situation user need to add accuracy status code and the HTTP response content to judge
+the web service is work well or not.
+
+HTTP Probe is a mechanism of probe pod is running correct or not. It is part of probe.
+Usually HTTP response is between 200 and 400 is correct
+
+This KEP is proposing add some extra info to check HTTP service was work correct or not.
+
+## Motivation
+Sometimes people need a accurate status code check for the HTTP probe. Such as I run the
+ElasticSearch on the kubernetes. When there is a special request, which has a large result.
+It can trigger the the circuit breaker. It cause all request to the ElasticSearch return
+429 temporary.
+
+Another is some special program when it encounter a problem it will return 302 instead of 
+another code.
```

Comment:
neither of those sound like conditions a probe should consider successful

Reply:
Yes, 429 is not a successful response code. But I my case the response code should be ignore it. 

429 : Some special query will triger the elasticsearch return 429 code. But it will turn to avaliable after some seconds. Compare with return code 429 I need to monitor some more important metrics, such as server triger full gc. The all interface will hang and unavaliable.

302 : I my company some spacial app will triger fusing will return the 302 code, But it not the problem of this app. Restart the app will not solve any problem.